R/record_versions.R

```diff
@@ -41,15 +37,25 @@ record_versions <- function(
 #' @title Get the current versions of packages
 #' @export
 #' @keywords internal
-#' @description Get the current versions of packages in the repos.
+#' @description Get the current versions of packages in the repo.
 #' @return A data frame of packages with their current versions and hashes.
 #' @inheritParams record_versions
 get_current_versions <- function(
-  repos = "https://r-releases.r-universe.dev"
+  repo = "https://r-releases.r-universe.dev"
 ) {
-  out <- utils::available.packages(repos = repos)
-  out <- out[, c("Package", "Version", "MD5sum")]
-  out <- as.data.frame(out)
+  listing <- paste(
+    contrib.url(repos = repo, type = "source"),
+    "PACKAGES.json?fields=RemoteSha",
+    sep = "/"
+  )
+  out <- jsonlite::stream_in(
+    con = url(listing),
```

Comment:
When I've used `jsonlite::stream_in()` in the past, I've wrapped `url()` in `gzcon()`, so it'd be `gzcon(url(listing))` here. Theoretically the time saved by reducing transmission bytes dwarfs any extra processing for the compression.

Reply:
Nice! Done in 3655a8db9eb0013c5aad80f5f0fb1b5eb17cad1a